storage/badger/operation/prefix.go

```diff
@@ -137,3 +139,15 @@ func b(v interface{}) []byte {
 		panic(fmt.Sprintf("unsupported type to convert (%T)", v))
 	}
 }
+
+// d is a reverse of b
+func d[T uint64](bytes []byte) (ret T) {
+
+	// See https://stackoverflow.com/a/73382479
+	switch any(ret).(type) {
+	case uint64:
+		return T(binary.BigEndian.Uint64(bytes))
```

Comment:
```suggestion
		return T(binary.BigEndian.Uint64(bytes))
	default:
		panic(...)
```
To avoid hiding failures from unimplemented types, which currently would return the empty value of type T.

Reply:
Hmm, this uses generics which is a fixed list of types, I don't think unsupported type will compile

Reply:
Actually I think the problem is different than what I initially commented. We're type switching on the type of the return value, which is always `uint64` - not the input type, since the input is always `[]byte`. 

⚠️  So types for which decoding is not implemented would not pass through the switch statement and silently fail, they would be incorrectly interpreted as `uint64` and the function would return an incorrect value. 

For example, we can try passing in the unsupported type `Identifier` (https://github.com/onflow/flow-go/commit/6d33da2a1fb74f44bed7eaf35caf3748cdc3a538):
```go
id := unittest.IdentifierFixture()
fmt.Println(id)
enc := b(id)
dec := d(enc)
fmt.Println(dec)
```


**Suggestion:** Modify function `d` to either correctly handle all types supported by `b`, or explicitly fail for unsupported types. (It would also be great to add some documentation to these functions and/or give them more descriptive names.)

Reply:
We do not pass unsupported type - this function always accepts `[]byte` - and its the raw encoding, without the type information. Type information is inferred from the return type parameter. In your case there is no other type possible than `uint64` - and thats how `d` interprets the bytes, and thats the type of `dec`.
 I have added another supported type `byte` to `d` - so now when you use it code won't compile without explicitly providing the type which I hope will make it more usable.

Reply:
Adding more than one type parameter option works to fix this - sounds good 👍